### DISCUSSION TITLE
Creation of constraints in schema (Or indirect relationships)

### DISCUSSION THREAD

**Discussion by laurent512** [prisma/prisma#13500] [Q&A] [answered]
Hello, 

I have a question concerning the creation of "constraints" at in my scema . 

The story is the following : There are Users, who can create Payments. Payment goes from _one_ user to _one_ other. 
User can ask for payments refund, by creating a RefundRequest. 
A refund is issued by a user, and he can ask to be refunded for _one more more_ payment he issued himself. 
(Also, a payment can only be refund once)

This lead me to the schema below : 

```
model User {
  id               String          @id @default(cuid())
  balanceEur       Decimal
  paymentsIssued   Payments[]      @relation("paymentsIssued")
  paymentsReceived Payments[]      @relation("paymentsReceived")
  refundRequest    RefundRequest[]
  // ...
}

model Payments {
  id              String         @id @default(cuid())
  from            User           @relation("paymentsIssued", fields: [fromUserId], references: [id])
  fromUserId      String
  to              User           @relation("paymentsReceived", fields: [toUserId], references: [id])
  toUserId        String
  refundRequest   RefundRequest? @relation(fields: [refundRequestId], references: [id])
  refundRequestId String?
  // ...
}

model RefundRequest {
  id       String     @id @default(cuid())
  **payments Payments[]**
  issuer   User       @relation(fields: [userId], references: [id])
  userId   String
  // How to be sure that issuer of the RefundRequest is as well the "from" in Payments
}

```

Is there a possibility to ensure at database level that the **payments** in the RefundRequest model are issued by the _issuer_ of the refundrequest ?

**Comment by nurul3101** [comment, marked as answer]
Hey @laurent512 👋 
I am not sure if you could achieve this easily at database level but on application side you can achieve this by writing a query like this:
```typescript
import { PrismaClient } from '@prisma/client';

const prisma = new PrismaClient();

async function main() {
  const userId = 'ck3q7qjq0z0q0101087jq0z0';

  const refundRequest = await prisma.refundRequest.findFirst({
    where: {
      userId,
    },
    include: {
      payments: {
        where: {
          fromUserId: userId,
        },
      },
    },
  });
}

main()
  .catch((e) => {
    throw e;
  })
  .finally(async () => {
    await prisma.$disconnect();
  });

```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
